types/ui/searchFilter.d.ts

```diff
@@ -1 +1,75 @@
-declare class SearchFilter {}
+/**
+ * A controller class for managing a text input widget that filters the contents of some other UI element
+ * @see {@link Application}
+ *
+ */
+declare class SearchFilter {
+  /**
+   * @param inputSelector   - The CSS selector used to target the text input element.
+   * @param contentSelector - The CSS selector used to target the content container for these tabs.
+   * @param initial         - The initial value of the search query.
+   *                          (default: `''`)
+   * @param callback        - A callback function which executes when the filter changes.
+   * @param delay           - The number of milliseconds to wait for text input before processing.
+   *                          (default: `100`)
+   */
+  constructor({
+    inputSelector,
+    contentSelector,
+    initial,
+    callback,
+    delay
+  }: {
```

Comment:
In theory, the options are optional but I agree that it doesn't make sense to type it like that. If you don't pass any options, as soon as you call `bind`, you'll get an error. I did the same with `Tabs`. :+1: